### DISCUSSION TITLE
cannot process requests multiprocessing

### DISCUSSION THREAD

**Discussion by evinstar** [fastapi/fastapi#7482] [Questions] [answered]
Hello:
   I have trouble with the concurrent process requests in fastapi:
#### My actions
what i do in my api is sending a request to other api, processing the response, and returning the result after processed .

I'm doing a test,in which i send 10 requests to my api by a threadpoolexecutor, it is expected to perfoming in concurrent way. But I find the app process the requests synchronized.

#### Codes:
start command `uvicorn main:app --reload --workers 4`

main.app.py
```
async def get_info(name: str):
    response = {'keyword': name}
    start = time.time()
    print(f'start: {name}')    
    return await search(strip(name))
```

test.py
```
with ThreadPoolExecutor(10) as executor:
     future_to_url = {executor.submit(get, url):url for url in urls}
     for future in as_completed(future_to_url):
        url = future_to_url[future]
        try:
            data = future.result()
            # time_cost = data.get("time_cost", 'None')
        except Exception as e:
            logger.info(f'{e} url: {url}')
        else:
            # logging.info(f'result: {data}, url: {url}')
            logger.info(f'success: {url}: {time_cost}')
```

python:3.7

Appreciate a lot for any help.

**Comment by dbanty** [comment, marked as answer]
It would be ideal if you could provide a completely self-contained example that we can run and see the same results.

You say the requests are synchronized, I assume that means the first will complete before the second begins. This tells me that your endpoint is blocking the event loop. I assume that the `search` function you’re awaiting is what’s calling the other API. How are you calling that API? Are you using an async method like `httpx` and awaiting? Or using a blocking method like `requests`?

**Comment by evinstar** [comment]
> 
> 
> It would be ideal if you could provide a completely self-contained example that we can run and see the same results.
> 
> You say the requests are synchronized, I assume that means the first will complete before the second begins. This tells me that your endpoint is blocking the event loop. I assume that the `search` function you’re awaiting is what’s calling the other API. How are you calling that API? Are you using an async method like `httpx` and awaiting? Or using a blocking method like `requests`?

Thanks a lot, In my app, there is include some special data authorized. But your answer has inspired me a lot. As you say, I use the  blocking http lib that is requests in search function, that maybe the key problem I need to solve.

Thanks again for your help ^_^.

**Comment by evinstar** [comment]
solved
